File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenceIndicatorRepository.java
package uk.gov.justice.hmpps.prison.repository.jpa.repository;

import org.springframework.data.repository.CrudRepository;
import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenceIndicator;

public interface OffenceIndicatorRepository extends CrudRepository<OffenceIndicator, Long> {
    Long deleteByIndicatorCodeAndOffence_Code(String indicatorCode, String offenceCode);
}

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenceIndicatorRepository.kt
package uk.gov.justice.hmpps.prison.repository.jpa.repository

import org.springframework.data.repository.CrudRepository
import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenceIndicator

interface OffenceIndicatorRepository : CrudRepository<OffenceIndicator, Long> {
  fun deleteByIndicatorCodeAndOffenceCode(indicatorCode: String, offenceCode: String): Long
  fun existsByIndicatorCodeAndOffenceCode(indicatorCode: String, offenceCode: String): Boolean
}
